subprojects/ide-native/src/main/java/org/gradle/ide/visualstudio/VisualStudioSolution.java

```diff
@@ -52,9 +52,11 @@
     Set<VisualStudioProject> getProjects();
 
     /**
-     * The component that this solution represents.
+     * The name of the component associated with this solution.
+     *
+     * @since 4.6
      */
-    NativeComponentSpec getComponent();
+    String getComponentName();
```

Comment:
I don't think we can drop this until we move to a single-solution approach.  Otherwise, we still need to know the name of our component so we can name the task appropriately.  Might be other ways to do this, but I kinda feel like we should worry about when we add support for the new plugins.

Reply:
I've moved this to an internal interface so the only things that should be visible through the DSL are the files that can be modified.